assets/src/design-system/components/textArea/index.js

```diff
@@ -0,0 +1,206 @@
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import { Text } from '../typography';
+import { themeHelpers, THEME_CONSTANTS } from '../../theme';
+import { focusCSS } from '../../theme/helpers';
+import labelAccessibilityValidator from '../../utils/labelAccessibilityValidator';
+import useInputEventHandlers from '../../utils/useInputEventHandlers';
+
+const Container = styled.div`
+  position: relative;
+  width: 100%;
+  min-width: 100px;
+`;
+
+const Label = styled(Text)`
+  margin-bottom: 12px;
+`;
+
+const Hint = styled(Text)`
+  margin-top: 12px;
+  color: ${({ hasError, theme }) =>
+    theme.colors.fg[hasError ? 'negative' : 'tertiary']};
+`;
+
+const InputContainer = styled.div(
+  ({ focused, hasError, theme }) => css`
+    box-sizing: border-box;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    padding: 8px 12px;
+    border: 1px solid
+      ${theme.colors.border[hasError ? 'negativeNormal' : 'defaultNormal']};
+    border-radius: ${theme.borders.radius.small};
+    overflow: hidden;
+
+    ${focused &&
+    !hasError &&
+    css`
+      border-color: ${theme.colors.border.defaultActive};
+    `};
+
+    :focus-within {
+      ${focusCSS(theme.colors.border.focus)};
+    }
+  `
+);
+
+const StyledTextArea = styled.textarea(
+  ({ theme }) => css`
+    height: 100%;
+    width: 100%;
+    padding: 0;
+    background-color: inherit;
+    border: none;
+    outline: none;
+    color: ${theme.colors.fg.primary};
+    resize: none;
+
+    ${themeHelpers.expandPresetStyles({
+      preset: {
+        ...theme.typography.presets.paragraph[
+          THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+        ],
+      },
+      theme,
+    })};
+
+    :disabled {
+      color: ${theme.colors.fg.disable};
+      border-color: ${theme.colors.border.disable};
+    }
+
+    :active:enabled {
+      color: ${theme.colors.fg.primary};
+    }
+  `
+);
+
+const Counter = styled.div`
+  text-align: right;
+  align-self: flex-end;
+  span {
+    color: ${({ theme }) => theme.colors.fg.tertiary};
+  }
+`;
+
+export const TextArea = forwardRef(
+  (
+    {
+      className,
+      disabled,
+      hasError,
+      hint,
+      id,
+      label,
+      onBlur,
+      onFocus,
+      value,
+      showCount = false,
+      maxLength,
+      isIndeterminate = false,
+      ...props
+    },
+    ref
+  ) => {
+    const textAreaId = useMemo(() => id || uuidv4(), [id]);
+    const textAreaRef = useRef(null);
+    const [focused, setFocused] = useState(false);
+
+    const hasMaxLength = typeof maxLength === 'number';
+    const hasCounter = showCount && hasMaxLength;
+
+    const { handleBlur, handleFocus } = useInputEventHandlers({
+      forwardedRef: ref,
+      inputRef: textAreaRef,
+      focused,
+      setFocused,
+      onBlur,
+      onFocus,
+    });
+
+    let displayedValue = hasMaxLength
+      ? value?.substring(0, maxLength - 1)
+      : value;
```

Comment:
🤔 Does a textarea's `maxlength` attribute cut off the string automatically? If so, we may not need to edit the value here, right?

Reply:
Hm, I thought I tested it and it didn't work as expected. But perhaps it was an oversight and I did something wrong. Will test again 👍

Reply:
Awesome, thanks! Will replace it.